packages/bloc_test/test/bloc_bloc_test_test.dart

```diff
@@ -550,4 +561,30 @@ Alternatively, consider using Matchers in the expect of the blocTest rather than
       });
     });
   });
+
+  group('tearDown', () {
+    late int tearDownCallCount;
+    int? state;
+
+    setUp(() {
+      tearDownCallCount = 0;
+    });
+
+    tearDown(() {
+      expect(tearDownCallCount, equals(1));
+    });
+
+    blocTest<CounterBloc, int>(
+      'is called after the test is run',
+      build: () => CounterBloc(),
+      act: (bloc) => bloc.add(CounterEvent.increment),
+      verify: (bloc) {
+        state = bloc.state;
```

Comment:
Is it necessary to set the `state` variable here and then expect that it's correct in `blocTest.tearDown`? Maybe I'm missing something, but it doesn't seem related to testing the `tearDown` callback.

Reply:
the goal of this was to ensure that tearDown executes last. This isn't how you'd use `tearDown` normally -- it's more to ensure that tearDown is called last.

Reply:
ok, that makes sense - thanks for claryfing.